src/components/search/Filters/filters.tsx

```diff
@@ -204,20 +247,51 @@ const Filters = ({
           MenuProps={{ autoFocus: false }}
         >
           {/* select all sessions */}
-          <MenuItem value="select-all">
+          <MenuItem className="h-10 items-center" value="select-all">
             <Checkbox
-              checked={chosenSessions.length === academicSessions.length}
+              checked={
+                academicSessions.length > 0 &&
+                chosenSessions.length === academicSessions.length
+              }
               indeterminate={
                 chosenSessions.length !== academicSessions.length &&
-                chosenSessions.length !== 0
+                chosenSessions.length !== 0 &&
+                !(
+                  chosenSessions.length === recentSemesters.length &&
+                  chosenSessions.every((el) => recentSemesters.includes(el))
+                ) // select-all is not indeterminate when recent is checked
```

Comment:
I feel like the select all checkbox should be in the indeterminate state when recents is checked